pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -511,6 +512,39 @@ func (ds *Plugin) ListFederationRelationships(ctx context.Context, req *datastor
 	return resp, nil
 }
 
+// UpdateFederationRelationship updates the given federation relationship.
+// Attributes are only updated if the correspondent mask value is set to true.
+func (ds *Plugin) UpdateFederationRelationship(ctx context.Context, fr *datastore.FederationRelationship, mask *types.FederationRelationshipMask) (newFr *datastore.FederationRelationship, err error) {
+	if fr == nil {
+		return nil, status.Error(codes.InvalidArgument, "federation relationship is nil")
+	}
+
+	if fr.TrustDomain.IsZero() {
+		return nil, status.Error(codes.InvalidArgument, "trust domain is required")
+	}
+
+	if mask.BundleEndpointUrl && fr.BundleEndpointURL == nil {
+		return nil, status.Error(codes.InvalidArgument, "bundle endpoint URL is required")
+	}
+
+	if mask.BundleEndpointProfile {
+		switch fr.BundleEndpointProfile {
+		case datastore.BundleEndpointWeb:
+		case datastore.BundleEndpointSPIFFE:
+			if fr.EndpointSPIFFEID.IsZero() {
+				return nil, status.Error(codes.InvalidArgument, "bundle endpoint SPIFFE ID is required")
+			}
+		default:
+			return nil, status.Errorf(codes.InvalidArgument, "unknown bundle endpoint profile type: %q", fr.BundleEndpointProfile)
+		}
+	}
+
```

Comment:
Can this validation be unified with the create method? For example, we could refactor it into a separate function that takes the FederationRelationship and FederationRelationshipMask. The CreateFederationRelationship could pass the "all" mask into the function. What do you think?

Reply:
Sounds great! I'll refactor it.